nano/secure/ledger_set_confirmed.cpp

```diff
@@ -0,0 +1,181 @@
+}
+
+std::shared_ptr<nano::block> nano::ledger_set_confirmed::get (store::transaction const & transaction, nano::block_hash const & hash) const
+{
+	auto block = ledger.store.block.get (transaction, hash);
+	if (!block)
+	{
+		return nullptr;
+	}
+	auto info = ledger.store.confirmation_height.get (transaction, block->account ());
+	if (!info)
+	{
+		return nullptr;
+	}
+	return block->sideband ().height <= info.value ().height ? block : nullptr;
+}
+
+std::optional<nano::pending_info> nano::ledger_set_confirmed::get (store::transaction const & transaction, nano::pending_key const & key) const
+{
+	auto result = ledger.store.pending.get (transaction, key);
+	if (!result && !exists_or_pruned (transaction, key.hash))
+	{
+		return std::nullopt;
+	}
+	return result;
+}
+
+nano::block_hash nano::ledger_set_confirmed::head (store::transaction const & transaction, nano::account const & account) const
+{
+	auto info = ledger.store.confirmation_height.get (transaction, account);
+	if (!info)
+	{
+		return 0;
+	}
+	return info.value ().frontier;
+}
+
+uint64_t nano::ledger_set_confirmed::height (store::transaction const & transaction, nano::account const & account) const
+{
+	auto head_l = head (transaction, account);
+	if (head_l.is_zero ())
+	{
+		return 0;
+	}
+	return get (transaction, head_l)->sideband ().height;
+}
+
+auto nano::ledger_set_confirmed::receivable_end () const -> receivable_iterator
+{
+	return receivable_iterator{};
+}
+
+auto nano::ledger_set_confirmed::receivable_upper_bound (store::transaction const & transaction, nano::account const & account) const -> receivable_iterator
+{
+	return receivable_iterator{ transaction, *this, receivable_lower_bound (transaction, account.number () + 1, 0) };
+}
+
+auto nano::ledger_set_confirmed::receivable_upper_bound (store::transaction const & transaction, nano::account const & account, nano::block_hash const & hash) const -> receivable_iterator
+{
+	auto result = receivable_lower_bound (transaction, account, hash.number () + 1);
+	if (!result || result.value ().first.account != account)
+	{
+		return receivable_iterator{ transaction, *this, std::nullopt };
+	}
+	return receivable_iterator{ transaction, *this, result };
+}
+
+std::optional<std::pair<nano::pending_key, nano::pending_info>> nano::ledger_set_confirmed::receivable_lower_bound (store::transaction const & transaction, nano::account const & account, nano::block_hash const & hash) const
+{
+	auto result = ledger.store.pending.begin (transaction, { account, hash });
+	while (result != ledger.store.pending.end () && !exists (transaction, result->first.hash))
+	{
+		++result;
+	}
+	if (result == ledger.store.pending.end ())
+	{
+		return std::nullopt;
+	}
+	return *result;
+}
+
+std::optional<nano::block_hash> nano::ledger_set_confirmed::successor (store::transaction const & transaction, nano::qualified_root const & root) const
+{
+	if (root.previous ().is_zero ())
+	{
+		return ledger.store.block.successor (transaction, root.previous ());
```

Comment:
Is this supposed to be like this? `root.previous ()` is zero here. Seems like code duplication is leading to subtle bugs.

Reply:
This function isn't actually used anywhere in the code; it looks left over from when this PR used polymorphism where this implementation was created but never used or tested.  Since it's not used I deleted it rather than fixing / testing.